packages/fxa-settings/src/lib/channels/firefox.ts

```diff
@@ -380,10 +383,21 @@ export class Firefox extends EventTarget {
           });
         });
       }),
-      new Promise(
-        (resolve) =>
-          (timeout = setTimeout(resolve, DEFAULT_SEND_TIMEOUT_LENGTH_MS))
-      ),
+      // Ideally, we would detect WebChannel support instead of relying on a timeout.
+      // However, it's difficult to reliably detect all compatible environments —
+      // including Firefox Desktop, Android (Fenix), and iOS (FxA iOS) — especially
+      // when considering misconfigurations. For example, when using `localhost`
+      // with a browser configured to talk to production or stage servers,
+      // the WebChannel may be present but not able to communicate correctly.
+      // Because of this, we fall back to a short timeout to detect unresponsiveness.
+      new Promise((resolve) => {
+        timeoutId = window.setTimeout(() => {
```

Comment:
Is the explicit `window` here for anything in particular?

Reply:
To make typescript happy - setTimeout alone returns a Timeout object instead of a number as it would in a browser env. Alternative would have been to cast as number

Reply:
Sounds like TS think you are writing nodejs code :-/